frontend/src/services/GooglePlacesService.ts

```diff
@@ -40,19 +41,30 @@ class GooglePlacesService {
       });
   }
 
-  getGooglePlacesDetails(placeID: string): Promise<JSON> {
+  getRestaurantDetails(placeID: string): Promise<Record<string, unknown>> {
+    if (this.currRestaurants.has(placeID)) {
+      const restaurantDetails = this.currRestaurants.get(placeID) as Record<
+        string,
+        unknown
+      >;
+      return Promise.resolve(restaurantDetails);
+    }
+
     return axios
-      .get('https://maps.googleapis.com/maps/api/place/details/json?', {
+      .get("https://maps.googleapis.com/maps/api/place/details/json?", {
         params: {
           place_id: placeID,
           fields:
-            'url,formatted_phone_number,opening_hours,website,review,photos',
+            "url,formatted_phone_number,opening_hours,website,review,photos,business_status," +
+            "formatted_address,geometry,name,price_level,rating,user_ratings_total",
           key: process.env.GOOGLE_PLACES_API_KEY,
```

Comment:
Something I just discovered recently that we could do here is to change this to `process.env.REACT_APP_GOOGLE_PLACES_API_KEY` and add this key to our .env file. I believe `react-scripts` loads these environment variables for us already so we don't also need to list `dotenv` as a dependency!

https://stackoverflow.com/a/56668716

Reply:
Yes! This was the way I did it for integrating the google maps api too

Reply:
OH WOAH! That's handy 😅  I'll change it to that instead!